### PR TITLE
config: add werror  and warnings options

### PR DESCRIPTION
**Pull request by Jafaral**

### CODE REVIEW

File: src/h/config.h [hunk 1]
```diff
@@ -548,6 +548,8 @@
 #endif					/* IntBits */
 
 #if (WordBits == 64)
+   #define LINTFRMT "l"
```
Comment (Don-Ward) [inline src/h/config.h:551]: My first thought on seeing this line was "What's lint got to do with it?".  It was only later when looking at rmisc.r that I realised it meant "Long Integer".  C99 introduced the inttypes.h header with predefined printf format specifiers (PRId64 etc.)  Is it worth considering using that?

File: src/runtime/rwinsys.r [hunk 2]
```diff
@@ -42,6 +42,7 @@ void checkpollevent(){
 
 #else					/* Graphics */
 #if NT
+int iconx(int argc, char **argv);
```
Comment (Don-Ward) [inline src/runtime/rwinsys.r:45]: I don't think we'll get a compiler fussy enough to complain about the difference between char ** and char*[], but why take the risk? i.e. declare it as int iconx(int argc, char *argv[]);

**Review by Don-Ward — APPROVED** [review]
Looks good. Perhaps my suggestion to use intypes.h is better taken as a separate item -- it would lead to quite a few changes if done systematically.

File: src/h/config.h (hunk 1 above)
Comment (Jafaral) [reply src/h/config.h:551]: Given your comment below, no action is needed here then.

File: src/runtime/rwinsys.r (hunk 2 above)
Comment (Jafaral) [reply src/runtime/rwinsys.r:45]: Only on Windows I got this warning, and I copied the function prototype verbatim from imain.r as it is actually defined.

**Comment by Jafaral** [comment]
@Don-Ward , if you are OK with it then let us get in please.

**Pull request merged**